core/types/src/keys.rs

```diff
@@ -117,6 +148,16 @@ impl<ADDR, KIND> TryFrom<&[u8; 32]> for Key<ADDR, KIND, RistrettoPublic> {
     }
 }
 
+/// Attempt to create a public key from a compressed point, wrapping
+/// [`RistrettoPublic::try_from`]
+impl<ADDR, KIND> TryFrom<[u8; 32]> for Key<ADDR, KIND, RistrettoPublic> {
+    type Error = KeyError;
+
+    fn try_from(p: [u8; 32]) -> Result<Self, Self::Error> {
+        Self::try_from(&p)
```

Comment:
Optional/Equivalent: Since `Self::try_from` takes a slice, we might as well make this work on a slice
```suggestion
impl<ADDR, KIND> TryFrom<&[u8; 32]> for Key<ADDR, KIND, RistrettoPublic> {
    type Error = KeyError;

    fn try_from(p: &[u8; 32]) -> Result<Self, Self::Error> {
        Self::try_from(p)
```

Reply:
there's _also_ a from slice version (:138), it just happens that the const array version alleviates some type constraint struggles when overriding serialization methods